### PR TITLE
feat: CCIE-1662 hvm GitHub PAT support

### PR DESCRIPTION
**Pull request by dspadea**
<details open>
  <summary><a href="https://czi-tech.atlassian.net/browse/CCIE-1662" title="CCIE-1662" target="_blank">CCIE-1662</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
  <td>HVM Github PAT support</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Task
        Task
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>Done</td>
    </tr>
  </table>
</details>

---

* Add support for github PAT auth
* Fix some help text
* Add support for a hvmconfig file
* Fix bug installing non-happy github projects (works for `fogg` and `aws-oidc` now! 🚀  )

### CODE REVIEW

**Comment by dspadea** [comment]
Currently failing one of the checks because happy 101 isn't published as a release yet. Should work later on. Will check back later. Otherwise, I think this is ready for review.

**Comment by dspadea** [comment]
> Currently failing one of the checks because happy 101 isn't published as a release yet. Should work later on. Will check back later. Otherwise, I think this is ready for review.

This was fixed by upgrading the install-happy github action version for api. (Thanks for the pointer, Hayden!)

File: hvm/cmd/list-releases.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ import (
 	"github.com/chanzuckerberg/happy/shared/githubconnector"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+
+//    "github.com/chanzuckerberg/happy/hvm/config"
```
Comment (jakeyheath) [inline hvm/cmd/list-releases.go:10]: remove

File: hvm/config/config.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+type HvmConfig struct {
+
+    GithubPAT *string
+
+}
+
+
+func GetHvmConfig() (*HvmConfig, error){
+	home, err := os.UserHomeDir()
```
Comment (jakeyheath) [inline hvm/config/config.go:18]: can you install a golang formatter for your editor, it looks like you have some formatting to do on these line. Or just run `go fmt` on the project

**Review by jakeyheath — APPROVED** [review]
small nits

**Pull request merged**